### DISCUSSION TITLE
HttpClient port selection

### DISCUSSION THREAD

**Discussion by zbrag** [symfony/symfony#43407] [Q&A]
Hi,

I'm trying to use HttpClient to consume an api which doesn't run on port 80 or 443. I'm trying to find out if I'm doing something wrong, so far I'm not finding any configuration options to set a port, so I'm guessing it isn't possible at the moment?

merci

**Comment by michaoj** [comment]
Why don't you just add :portNumber to your uri ?

**Reply by michaoj** [reply]
And there is an option to set the port otherwise
https://symfony.com/doc/current/reference/configuration/framework.html#http-port

**Reply by stof** [reply]
@michaoj this config is about configuring the router, not the http client

**Reply by michaoj** [reply]
my bad, read the doc too fast :(

**Reply by zbrag** [reply]
> Why don't you just add :portNumber to your uri ?

I've tried that, but it seems Symfony ignores the port, see for example https://github.com/symfony/http-client/blob/5.3/NativeHttpClient.php#L120 and https://github.com/symfony/http-client/blob/5.3/CurlHttpClient.php#L189 , or am I missing something else?
